src/clj_puppetdb/vcr.clj

```diff
@@ -0,0 +1,28 @@
+(ns clj-puppetdb.vcr
+  (:require [clojure.edn :as edn]
+            [me.raynes.fs :as fs]
+            [puppetlabs.kitchensink.core :refer [utf8-string->sha1]]
+            [puppetlabs.http.client.sync :as http]
+            [puppetlabs.kitchensink.core :refer [dissoc-in]]))
+
+(defn- vcr-filename
+  [vcr-dir path]
+  (str vcr-dir "/" (utf8-string->sha1 path) ".clj"))
```

Comment:
Ah, I didn't realize the query is already part of the path. As you rightly point out the query params are sorted too - though the nested structures are not, but that's probably a corner case.